python/cudf/cudf/_lib/pylibcudf/scalar.pyx

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+cimport pyarrow.lib
+from cython cimport no_gc_clear
+from libcpp.memory cimport unique_ptr
+
+import pyarrow.lib
+
+from rmm._lib.memory_resource cimport get_current_device_resource
+
+from cudf._lib.cpp.scalar.scalar cimport fixed_point_scalar, scalar
+from cudf._lib.cpp.wrappers.decimals cimport (
+    decimal32,
+    decimal64,
+    decimal128,
+    scale_type,
+)
+
+from .types cimport DataType, type_id
+
+
+# The DeviceMemoryResource attribute could be released prematurely
+# by the gc if the DeviceScalar is in a reference cycle. Removing
+# the tp_clear function with the no_gc_clear decoration prevents that.
+# See https://github.com/rapidsai/rmm/pull/931 for details.
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** ‏Perhaps we should (in a followup) document this issue more discoverably by adding some text to the dev docs.

Reply:
Yeah perhaps there's a more discoverable place for this, @shwina WDYT?